code/game/objects/items/devices/scanners/mining.dm

```diff
@@ -1,5 +1,9 @@
-/turf/simulated
-	var/surveyed
+/datum/extension/buried_resources
+	base_type = /datum/extension/buried_resources
+	expected_type = /turf
+	flags = EXTENSION_FLAG_IMMEDIATE
```

Comment:
Does it really need immediate flag? It's put on a lot of places, and doesn't really do anything on its own, just set/read by others. Might shave few KB if you leave it lazy